src/mixing_fab.h

```diff
@@ -0,0 +1,158 @@
+
+#ifndef CYCAMORE_SRC_MIXING_FAB_H_
+#define CYCAMORE_SRC_MIXING_FAB_H_
+
+#include <string>
+#include "cyclus.h"
+#include "cycamore_version.h"
+
+namespace cycamore {
+  
+  /// MixingFab takes in N streams of material and mixes them accordingly
+  /// to the ratios prodided by the user.
+  ///
+  /// The MixingFab has N input inventories: one for each streams to be mixed,
+  /// and one output stream.
+  ///
+  /// The supplying of mixed material is constrained by available inventory of
+  /// mixed material quantities.
+  /// @endcode
+  
+  
+class MixingFab : public cyclus::Facility {
+#pragma cyclus note {   	  \
+    "niche": "mixing facility",				  \
+    "doc":								\
+    " the archetype basically mixes N streams with fixed, static,"\
+    "user-specified ratios into a single output stream" \
+    "\n\n " \
+    "The MixingFab has N input inventories: one for each streams to be mixed,"\
+    "and one output stream."\
+    "\n\n" \
+    "The supplying of mixed material is constrained by available inventory of"\
+    "mixed material quantities.", \
+    }
+  
+  friend class MixingFabTest;
+
+  public:
+  MixingFab(cyclus::Context* ctx);
+  virtual ~MixingFab() {};
+  
+  virtual void Tick();
+  virtual void Tock(){};
+  virtual void EnterNotify();
+  
+  virtual void AcceptMatlTrades(const std::vector<std::pair<
+                                cyclus::Trade<cyclus::Material>, cyclus::Material::Ptr> >& responses);
+  
+  virtual std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+      GetMatlRequests();
+
+
+  #pragma cyclus clone
+  #pragma cyclus initfromcopy
+  #pragma cyclus infiletodb
+  #pragma cyclus initfromdb
+  #pragma cyclus schema
+  #pragma cyclus annotations
+  #pragma cyclus snapshot
+  // the following pragmas are ommitted and the functions are written
+  // manually in order to handle the vector of resource buffers:
+  //
+  //     #pragma cyclus snapshotinv
+  //     #pragma cyclus initinv
+  
+  virtual cyclus::Inventories SnapshotInv();
+  virtual void InitInv(cyclus::Inventories& inv);
+
+  
+  protected:
+    
+    
+  #pragma cyclus var { \
+    "doc": "Ordered list of commodities on which to requesting stream material, each commodity corresponds to a input stream.", \
+    "uilabel": "Stream Commodities", \
+    "uitype": ["oneormore", "incommodity"], \
+  }
+  std::vector<std::string> in_commods;
+  #pragma cyclus var { \
+    "default": [], \
+    "uilabel": " Stream Preferences", \
+    "doc": "Stream commodity request preferences for each of the given commodities (same order)." \
+    " If unspecified, default is to use 1.0 for all preferences.", \
+  }
+  std::vector<double> fill_commod_prefs;
+  
+  #pragma cyclus var { \
+  "doc": "Size of each material stream inventory.", \
```

Comment:
``... input material stream inventory - i.e. the quantity of each stream type to keep on-hand)``